src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1142,15 +1137,20 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               def longWidened = tpSym == LongClass && targetIsWide
               intWidened || longWidened
             }
-            if (isInharmonic)
+            if (isInharmonic) {
               // not `context.deprecationWarning` because they are not buffered in silent mode
-              context.warning(tree.pos, s"Widening conversion from ${tpSym.name} to ${ptSym.name} is deprecated because it loses precision. Write `.to${ptSym.name}` instead.", WarningCategory.Deprecation)
-            else {
+              val msg = s"Widening conversion from ${tpSym.name} to ${ptSym.name} is deprecated because it loses precision. Write `.to${ptSym.name}` instead."
+              val orig = currentUnit.sourceAt(tree.pos)
+              context.warning(tree.pos, msg, WarningCategory.Deprecation,
+                runReporting.codeAction("add conversion", tree.pos, if (orig.contains(" ")) s"($orig).to${ptSym.name}" else s"$orig.to${ptSym.name}", msg))
+            } else {
               object warnIntDiv extends Traverser {
                 def isInt(t: Tree) = ScalaIntegralValueClasses(t.tpe.typeSymbol)
                 override def traverse(tree: Tree): Unit = tree match {
                   case Apply(Select(q, nme.DIV), _) if isInt(q) =>
-                    context.warning(tree.pos, s"integral division is implicitly converted (widened) to floating point. Add an explicit `.to${ptSym.name}`.", WarningCategory.LintIntDivToFloat)
+                    val msg = s"integral division is implicitly converted (widened) to floating point. Add an explicit `.to${ptSym.name}`."
+                    context.warning(tree.pos, msg, WarningCategory.LintIntDivToFloat,
+                      runReporting.codeAction("add conversion", tree.pos, s"(${currentUnit.sourceAt(tree.pos)}).to${ptSym.name}", msg))
```

Comment:
This is the one where we make it (too) easy to make a bad choice. If they really want it, they can do it the old-fashioned way. (Tell the intern to edit all the conversions.)